istioctl/cmd/waypoint.go

```diff
@@ -145,6 +151,84 @@ func waypointCmd() *cobra.Command {
 			return nil
 		},
 	}
+	waypointListCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List managed waypoint configurations",
+		Long:  "List managed waypoint configurations in the cluster created by istioctl",
+		Example: `  # List all waypoints in the cluster
+  # List all waypoints in a specific namespace
+  istioctl x waypoint list --namespace default
+
+  # List all waypoints in the cluster
+  istioctl x waypoint list -A`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			writer := cmd.OutOrStdout()
+			client, err := kubeClient(kubeconfig, configContext)
+			if err != nil {
+				return fmt.Errorf("failed to create Kubernetes client: %v", err)
+			}
+			ns := defaultNamespace
+			if allNamespaces {
+				ns = ""
+			} else if namespace != "" {
+				ns = namespace
+			}
+			gws, err := client.GatewayAPI().GatewayV1beta1().Gateways(ns).
+				List(context.Background(), metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			if len(gws.Items) == 0 {
+				fmt.Fprintln(writer, "No waypoints found.")
+				return nil
+			}
+			w := new(tabwriter.Writer).Init(writer, 0, 8, 5, ' ', 0)
+			slices.SortFunc(gws.Items, func(i, j gateway.Gateway) bool {
+				if i.Namespace <= j.Namespace {
+					return true
+				}
+				if i.Namespace > j.Namespace {
+					return false
+				}
+				return i.Name < j.Name
+			})
+			filteredGws := make([]gateway.Gateway, 0)
+			for _, gw := range gws.Items {
+				expectedName := makeGatewayName(gw.Annotations[constants.WaypointServiceAccount])
```

Comment:
should we also to make sure these gateway's have the waypoint class name? `spec.GatewayClassName: istio-waypoint`.  Not sure if the ServiceAccount annotation will be enough in the long term

Reply:
Good point, thanks